frontend/public/components/dashboard/dashboards-page/cluster-dashboard/health-item.tsx

```diff
@@ -28,29 +37,69 @@ import { uniqueResource } from './utils';
 import { getPrometheusQueryResponse } from '../../../../actions/dashboards';
 import { ClusterDashboardContext } from './context';
 
+const OperatorRow: React.FC<OperatorRowProps & {
+  LoadingComponent: () => JSX.Element;
+  Component: React.ComponentType<OperatorRowProps> | LazyLoader<OperatorRowProps>;
+  key: string;
+  isResolved: boolean;
+}> = ({ operatorStatus, isResolved, key, Component, LoadingComponent }) => {
+  const ResolvedComponent = Component as React.ComponentType<OperatorRowProps>;
+  return isResolved ? (
```

Comment:
The unresolved vs. resolved component duality causes more complexity, we should refactor the code to reduce the complexity.